File: desktop/web/src/ipc/community.ts
import { invoke } from '@tauri-apps/api/core'
import type { TemplateMeta } from '../types/community'
import type { CanopiFile } from '../types/design'

export async function getTemplateCatalog(): Promise<TemplateMeta[]> {
  return invoke('get_template_catalog')
}

export async function getTemplatePreview(id: string): Promise<TemplateMeta> {
  return invoke('get_template_preview', { id })
}

export async function acquireDesignTemplate(id: string): Promise<CanopiFile> {
  return invoke('acquire_design_template', { id })
}

File: desktop/src/commands/community.rs
pub use crate::services::community::TemplateMeta;
use common_types::design::CanopiFile;

#[tauri::command]
pub fn get_template_catalog() -> Result<Vec<TemplateMeta>, String> {
    Ok(crate::services::community::get_template_catalog())
}

#[tauri::command]
pub fn get_template_preview(id: String) -> Result<TemplateMeta, String> {
    crate::services::community::get_template_preview(&id)
}

#[tauri::command]
pub async fn acquire_design_template(id: String) -> Result<CanopiFile, String> {
    crate::blocking::run_blocking("template acquisition", move || {
        crate::services::community::acquire_template_blocking(id)
    })
    .await
}
